File: kAndr/ch_1/reverse.py
# reverses the character string s
def reverse(s):
    reversed = ""

    # create new string by appending the characters of s in reverse order
    current_idx = len(s) - 1
    while current_idx >= 0:
        reversed += s[current_idx]
        current_idx -= 1
    return reversed

print(reverse("hello world"))

File: kAndr/ch_1/reverse.c
#include <stdio.h>
#include <string.h>

void print_string(char *string, int length) {
    int i;
    printf("string output: ");
    for (i = 0; i <= length; i++) {
        printf("%c", string[i]);
    }
    printf("\n");
}

void reverse(char *s, char *reverse, int length) {
    int i; // start at last character
    int j = 0; // reversed string index

    for (i = length; i >= 0; i--) {
        reverse[j] = s[i];
        j++;
    }
}


int main() {
    int length;
    char *s = "hello world";

    length = strlen(s) - 1;
    char r[length];

    reverse(s, r, length);
    print_string(r, length);
}